src/data/model/LayerTreeNode.js

```diff
@@ -40,6 +40,14 @@ Ext.define('GeoExt.data.model.LayerTreeNode', {
                     return true;
                 }
             }
+        }, {
+            /**
+             * @private
+             * This should be set via tree panel.
+             */
+            name: 'toggleMode',
```

Comment:
I suggest using a more 'internal' and private-sounding name; perhaps '_gx_toggleMode'?